File: hw/nase/u3/hw24CompositePattern/CompossitePattern.javaScript/model/Employee.js
//Jennyfer Nase

export class Employee {
    constructor() {
        if (this.constructor === Employee) {
            throw new Error("Abstract class 'Employee' cannot be instantiated directly.");
        }
        this.name = "not assigned yet";
        this.title = "not assigned yet";
    }

    stateName() {
        console.log(`${this.title} ${this.name}`);
    }
}

File: hw/nase/u3/hw24CompositePattern/CompossitePattern.java/Employee/src/main/java/espe/edu/ec/employee/model/Employee.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package espe.edu.ec.employee.model;

/**
 *
 * @author Jennyfer Nase
 */
public abstract class Employee {
    protected String name = "not assigned yet";
    protected String title = "not assigned yet";

    public void stateName() {
        System.out.println(title + " " + name);
    }
}